modules/openglqt/src/processors/canvasprocessorwidgetqt.cpp

```diff
@@ -99,118 +106,128 @@ CanvasProcessorWidgetQt::CanvasProcessorWidgetQt(Processor* p)
+        canvas_->resize(static_cast<int>(logicalDim.x), static_cast<int>(logicalDim.y));
+
+        Super::setVisible(false);
     }
 
-    nameChange_ =
-        processor_->onDisplayNameChange([this](std::string_view newName, std::string_view) {
-            setWindowTitle(utilqt::toQString(newName));
-            RenderContext::getPtr()->setContextName(canvas_->contextId(), newName);
-        });
+    setWindowFlag(Qt::WindowStaysOnTopHint, CanvasProcessorWidget::isOnTop());
+    connect(qApp, &QApplication::applicationStateChanged, this, [this](Qt::ApplicationState state) {
+        utilqt::setOnTop(this, isOnTop() && state == Qt::ApplicationActive);
+    });
 
-    canvas_->setVisible(ProcessorWidget::isVisible());
     {
         // ignore internal state updates, i.e. position, when showing the widget
         // On Windows, the widget hasn't got a decoration yet. So it will be positioned using the
         // decoration offset, i.e. the "adjusted" position.
         util::KeepTrueWhileInScope ignore(&ignoreEvents_);
-        QWidget::setVisible(ProcessorWidget::isVisible());
+        Super::setVisible(ProcessorWidget::isVisible());
     }
     RenderContext::getPtr()->activateDefaultRenderContext();
 
     installEventFilter(new utilqt::WidgetCloseEventFilter(this));
 }
 
-CanvasProcessorWidgetQt::~CanvasProcessorWidgetQt() {}
+CanvasProcessorWidgetQt::~CanvasProcessorWidgetQt() = default;
 
 void CanvasProcessorWidgetQt::setVisible(bool visible) {
-    if (visible) {
-        canvas_->show();
-    } else {
-        canvas_->hide();
+    if (Super::isVisible() != visible) {
+        Super::setVisible(visible);  // This will trigger show/hide events.
+    }
+}
+
+void CanvasProcessorWidgetQt::setFullScreen(bool fullScreen) {
+    if (fullScreen != Super::isFullScreen()) {
+        utilqt::setFullScreen(this, fullScreen);
     }
-    QWidget::setVisible(visible);  // This will trigger show/hide events.
+    CanvasProcessorWidget::setFullScreen(fullScreen);
 }
 
-void CanvasProcessorWidgetQt::show() { CanvasProcessorWidgetQt::setVisible(true); }
-void CanvasProcessorWidgetQt::hide() { CanvasProcessorWidgetQt::setVisible(false); }
+void CanvasProcessorWidgetQt::setOnTop(bool onTop) {
+    utilqt::setOnTop(this, onTop);
+    CanvasProcessorWidget::setOnTop(onTop);
+}
 
 void CanvasProcessorWidgetQt::setPosition(ivec2 pos) {
-    if (pos != utilqt::toGLM(QWidget::pos())) {
-        QWidget::move(pos.x, pos.y);  // This will trigger a move event.
+    if (pos != utilqt::toGLM(Super::pos())) {
+        Super::move(pos.x, pos.y);  // This will trigger a move event.
     }
 }
 
 void CanvasProcessorWidgetQt::setDimensions(ivec2 dimensions) {
-    if (dimensions != utilqt::toGLM(QWidget::size())) {
-        QWidget::resize(dimensions.x, dimensions.y);  // This will trigger a resize event.
+    if (dimensions != utilqt::toGLM(Super::size())) {
+        const auto dpr = window()->devicePixelRatio();
```

Comment:
not sure devicePixelRatio seems quite descriptive? or how do you mean? A question here is what the processor widget size mean, is that the size of the widget of the size of the canvas/fbo.

Reply:
I thought that this part was the same as in propagateResizeEvent, but that one is the reverse.
We should probably mention this scaling in the canvasprocessorwidgetqt.h as it might be confusing that the dimension does not end up what one specified.

Reply:
Well what does the "canvas size" in the canvas processor mean? is that the widget size in logical pixels or the fob size in physical pixels?

Reply:
for the canvas processor I use the "canvas size" to mean the fbo size, but for the new canvas with properties, that makes little sense, and I only show the widget size